### PR TITLE
Metadata: No extra properties

### LINKED ISSUE #30169: Fail metadata validation on extraneous fields
* Disable top-level extra fields
  * allow-list known properties
 
Top-level allowExtra was enabled when fighting with underscores.

What do we do about old files with `_ab...`.  Might need special-case handing. We could store them in the allow-list for historical reasons.  These problems go away with V2 of metadata spec.   Could we remove those old versions from the registries?
  * let's remove old metadata versions them! 
  * These can be identified by metadata entries with `_ab_.**` fields
     * we can find them by certain dates... (PLAN A)
     * could we delete everything before this date?  Seems scary... but... maybe? (PLAN B)

### PR DESCRIPTION
**Pull request by bnchrch**
## Overview
Disables extra properties from metadata

closes #30169 

## Prior to merge
- [x] update all versioned metadata files in production bucket to have `ab_internal` instead of `_ab_internal`
- [x] Run validate --all on all metadata files to catch any incorrect fields

### CODE REVIEW

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/AirbyteInternal.py [hunk 1]
```diff
@@ -1,7 +1,3 @@
-#
-# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
```
Comment (bnchrch) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/AirbyteInternal.py:2]: @alafanechere Ive never asked, what is the cannonical way to add these?

Comment (erohmensing) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/AirbyteInternal.py:2]: I think the formatter adds them? But the formatter only formats metadata ci code when it runs on other steps that usually get skipped when running airbyte-ci only

Comment (erohmensing) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/AirbyteInternal.py:2]: how'd they end up removed?

File: airbyte-ci/connectors/metadata_service/lib/tests/fixtures/metadata_validate/invalid/metadata_extra_data.yaml
Comment (erohmensing) [inline airbyte-ci/connectors/metadata_service/lib/tests/fixtures/metadata_validate/invalid/metadata_extra_data.yaml:1]: Gotta love unit tests telling you exactly what the  behavior is (even if what is expected changes) 😄

**Review by erohmensing — APPROVED** [review]
Was looking for a "caught one!" fishing gif to refer to the single invalid metadata file. Alas, no cigar

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/AirbyteInternal.py (hunk 1 above)
Comment (bnchrch) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/AirbyteInternal.py:2]: When you regenerate the models it doesnt have any context about modifications we made after the last generation.

So essentially it removes the copyright

**Pull request merged**